packages/vue-app/template/store.js

```diff
@@ -3,73 +3,86 @@ import Vuex from 'vuex'
-    const namePath = name.split(/\//)
+    for (const filename of filenames) {
+      let name = filename.replace(/^\.\//, '').replace(/\.(<%= extensions %>)$/, '')
+      if (name === 'index') continue
 
-    name = namePath[namePath.length - 1]
-    if (['state', 'getters', 'actions', 'mutations'].includes(name)) {
-      const module = getModuleNamespace(storeData, namePath, true)
-      appendModule(module, filename, name)
-      continue
-    }
+      const namePath = name.split(/\//)
 
-    // If file is foo/index.js, it should be saved as foo
-    const isIndex = (name === 'index')
-    if (isIndex) {
-      namePath.pop()
-    }
+      name = namePath[namePath.length - 1]
+      if (['state', 'getters', 'actions', 'mutations'].includes(name)) {
+        const module = getModuleNamespace(storeData, namePath, true)
+        appendModule(module, filename, name)
+        continue
+      }
 
-    const module = getModuleNamespace(storeData, namePath)
-    const fileModule = getModule(filename)
+      // If file is foo/index.js, it should be saved as foo
+      const isIndex = (name === 'index')
+      if (isIndex) {
+        namePath.pop()
+      }
 
-    name = namePath.pop()
-    module[name] = module[name] || {}
+      const module = getModuleNamespace(storeData, namePath)
+      const fileModule = getModule(filename)
 
-    // if file is foo.js, existing properties take priority
-    // because it's the least specific case
-    if (!isIndex) {
-      module[name] = Object.assign({}, fileModule, module[name])
-      module[name].namespaced = true
-      continue
-    }
+      name = namePath.pop()
+      module[name] = module[name] || {}
 
-    // if file is foo/index.js we want to overwrite properties from foo.js
-    // but not from appended mods like foo/actions.js
-    const appendedMods = {}
-    if (module[name].appends) {
-      appendedMods.appends = module[name].appends
-      for (const append of module[name].appends) {
-        appendedMods[append] = module[name][append]
+      // if file is foo.js, existing properties take priority
+      // because it's the least specific case
+      if (!isIndex) {
+        module[name] = Object.assign({}, fileModule, module[name])
+        module[name].namespaced = true
+        continue
+      }
+
+      // if file is foo/index.js we want to overwrite properties from foo.js
+      // but not from appended mods like foo/actions.js
+      const appendedMods = {}
+      if (module[name].appends) {
+        appendedMods.appends = module[name].appends
+        for (const append of module[name].appends) {
+          appendedMods[append] = module[name][append]
+        }
       }
-    }
 
-    module[name] = Object.assign({}, module[name], fileModule, appendedMods)
-    module[name].namespaced = true
+      module[name] = Object.assign({}, module[name], fileModule, appendedMods)
+      module[name].namespaced = true
+    }
+    // If the environment supports hot reloading...
+    if (module.hot) {
```

Comment:
We could guard that code with a lodash template and check for prod to remove the extra bytes I guess 🤔

Reply:
Agree, removing dead code is smart, I don't know if webpack will automatically remove it, but we can use lodash template to directly remove this part :)

Same for the server, we could also use `if (process.client && module.hot)` around a `<% if (isDev) {%>`